app/src/main/java/org/wikipedia/search/RecentSearchDatabaseTable.kt

```diff
@@ -40,11 +40,13 @@ class RecentSearchDatabaseTable : DatabaseTable<RecentSearch>(SearchHistoryContr
     }
 
     override fun onUpgradeSchema(db: SQLiteDatabase, fromVersion: Int, toVersion: Int) {
-        db.execSQL("CREATE TABLE IF NOT EXISTS " + SearchHistoryContract.TABLE +
-                " ( " + getColumnsAdded(dBVersionIntroducedAt).joinToString(", ") + " )")
+        if (toVersion == DB_VER_FIXED_FROM_CORRUPTION) {
```

Comment:
We don't want to attempt recreating the table on *every* upgrade, just the one that rescues it from corruption.